File: homework4/Circle.cpp
#include "Circle.h"

Circle::Circle() {

}
Circle::Circle(double newRadius) {
	this->radius = newRadius;
}
Circle::Circle(double newRadius, string newColor, bool fill) {
	this->radius = newRadius;
	Shape::setColor(newColor);
	Shape::setFilled(fill);
}

void Circle::setRadius(double newRadius) {
	this->radius = newRadius;
}
double Circle::getRadius() const {
	return radius;
}

double Circle::getPerimeter() const {
	return (2 * getRadius()) * 3.14159;
}
double Circle::getArea() const {
	return (getRadius() * getRadius()) * 3.14159;
}

string Circle::toString() const {
	stringstream ss;
	ss << "circle, " << getRadius() << ", ";

	return ss.str();
}

File: GeometricObject_Inheritance/Circle.py
from GeometricObject import GeometricObject
import math

class Circle(GeometricObject):
    def __init__(self, radius, color, filled):
        self.__radius = radius
        self.setColor(color)
        self.setFilled(filled)

    def getRadius(self):
        return self.__radius

    def setRadius(self, radius):
        self.__radius = radius

    def getArea(self):
        return self.getRadius() * self.getRadius() * math.pi

    def getDiameter(self):
        return self.getRadius() * 2

    def getPerimeter(self):
        return self.getRadius() * 2 * math.pi


